packages/peregrine/lib/talons/SignIn/useSignIn.js

```diff
@@ -56,10 +56,20 @@ export const useSignIn = props => {
                 await setToken(token);
                 await getUserDetails({ fetchUserDetails });
 
-                // Then remove the old, guest cart and get the cart id from gql.
+                // Then remove the old guest cart and get the cart id from gql.
                 // TODO: This logic may be replacable with mergeCart in 2.3.4
                 await removeCart();
 
+                // Delete any cached cart data.
+                Object.keys(client.cache.data.data).forEach(key => {
+                    if (key.match(/^Cart/)) {
+                        // Delete the cache entry
+                        client.cache.data.delete(key);
```

Comment:
If this approach works we will need to apply it to the `placeOrder` handler in #2288